kolibri_server_setup.py

```diff
@@ -161,64 +167,60 @@ def save_nginx_conf_port(port, nginx_conf=None):
+        "  }}\n\n"
+        "  location {path_prefix} {{\n"
+        "    include uwsgi_params;\n"
+        "    uwsgi_pass unix:///tmp/kolibri_uwsgi.sock;\n"
+        "    proxy_ignore_headers Vary;\n"
+        "    error_page 502 @error502;\n"
+        "  }}\n\n"
+        "  location @error502 {{\n"
+        "    include uwsgi_params;\n"
+        "    uwsgi_pass unix:///tmp/kolibri_hashi_uwsgi.sock;\n"
+        '    proxy_ignore_headers Vary;\n'
+        "  }}\n"
+        "}}\n"
+    ).format(port=port, path_prefix=path_prefix)
 
-    configuration = "\nlisten {};\n".format(port)
-
-    with open(nginx_conf, "a") as nginx_conf_file:
+    with open(nginx_conf, "w") as nginx_conf_file:
         nginx_conf_file.write(configuration)
 
 
-def save_nginx_conf_include(static_root, nginx_conf=None):
+def save_nginx_conf_include(zip_port):
     """
     Automatically writes the dynamic Nginx configuration include from Kolibri
     configuration.
 
     This function is called from within the DJANGO_SETTINGS_MODULE after it
-    has defined STATIC_ROOT.
-
-    The nginx configuration snippet is loaded with in the main Nginx
-    configuration which contains::
-
-        include /path/to/.kolibri/nginx.conf;
-
-    Note that because we cannot load django.conf.settings from this module
-    (because Django settings depend on Kolibri conf), we have a sort of
-    circular issue with STATIC_ROOT. We cannot fetch settings.STATIC_ROOT
-    unless we start writing the Nginx configuration elsewhere during the load
-    order.
+    has defined ZIP_CONTENT_PORT and URL_PATH_PREFIX.
     """
 
-    if nginx_conf is None:
-        nginx_conf = os.path.join(KOLIBRI_HOME, "nginx.conf")
+    nginx_conf = os.path.join(KOLIBRI_HOME, "nginx.conf")
 
-    with open(nginx_conf, "w") as nginx_conf_file:
-        configuration = (
-            "# This file is maintained AUTOMATICALLY and will be overwritten\n"
-            "#\n"
-            "# Do not edit this file. If you are using the kolibri-server"
-            "package,\n"
-            "# please write custom configurations in /etc/kolibri/nginx.d/\n"
-            "\n"
-            "location {path_prefix}static {{\n"
-            "    alias  {static_dir};\n"
-            "    expires 365d;\n"
-            '    add_header Cache-Control "public";\n'
-            "}}\n"
-            "location {path_prefix}content {{\n"
-            "    alias  {content_dir}/;\n"
-            "    expires 365d;\n"
-            '    add_header Cache-Control "public";\n'
-            "}}\n"
-        ).format(
-            static_dir=static_root,
-            content_dir=get_content_dir_path(),
-            path_prefix=path_prefix,
-        )
+    configuration = (
+        "\n"
+        "server{{\n"
+        "  listen {zip_port};\n"
+        "  location {path_prefix} {{\n"
+        "    include uwsgi_params;\n"
+        "    uwsgi_pass unix:///tmp/kolibri_hashi_uwsgi.sock;\n"
+        '    proxy_ignore_headers Vary;\n'
```

Comment:
Wondering if we could just consolidate the nginx configuration and use the same for each? No harm in having the 502 handling in the zipcontent configuration.

Reply:
It's even simpler, after so many changes from 0.14.x a refactor allows cleaning more code